instrumentation/servlet/servlet-3.0/library/src/main/java/io/opentelemetry/instrumentation/servlet/v3_0/OpenTelemetryServletFilter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.servlet.v3_0;
+
+import static io.opentelemetry.instrumentation.servlet.v3_0.copied.Servlet3Singletons.FILTER_MAPPING_RESOLVER;
+
+import io.opentelemetry.instrumentation.servlet.v3_0.copied.CallDepth;
+import io.opentelemetry.instrumentation.servlet.v3_0.copied.Servlet3FilterMappingResolverFactory;
+import io.opentelemetry.instrumentation.servlet.v3_0.copied.Servlet3RequestAdviceScope;
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * OpenTelemetry Library instrumentation for Java Servlet based applications that can't use a Java
+ * Agent. Due to inherit limitations in the servlet filter API, instrumenting at the filter level
+ * will miss anything that happens earlier in the filter stack or problems handled directly by the
+ * app server. For this reason, Java Agent instrumentation is preferred when possible.
+ */
+@WebFilter("/*")
+public class OpenTelemetryServletFilter implements Filter {
+
+  @Override
+  public void init(FilterConfig filterConfig) {
+    FILTER_MAPPING_RESOLVER.set(this, new Servlet3FilterMappingResolverFactory(filterConfig));
```

Comment:
is there value in using a virtual field for this instead of just adding a field to this class? or maybe this falls under "I tried to keep the instrumentation as similar as possible to the javaagent instrumentation."